dali/python/nvidia/dali/plugin/tf.py

```diff
@@ -116,5 +117,136 @@ def DALIRawIterator():
     return _dali_tf
 
 
+def _get_tf_minor_version():
+  return tf.__version__.split('.')[1]
+
+
+if _get_tf_minor_version() in {'13', '14'}:
+  from tensorflow.python.framework import ops
+  from tensorflow.python.data.ops import dataset_ops
+  from tensorflow.python.data.util import structure
+  import functools
+
+  class _DALIDatasetV2(dataset_ops.DatasetSource):
+    def __init__(
+      self,
+      pipeline = '',
+      batch_size = 1,
+      num_threads = 4,
+      device_id = 0,
+      exec_separated = False,
+      prefetch_queue_depth = 2,
+      cpu_prefetch_queue_depth = 2,
+      gpu_prefetch_queue_depth = 2,
+      shapes = [], 
+      dtypes = []):
+
+      assert(len(shapes) == len(dtypes),
+        "Different number of provided shapes and dtypes.")
+
+      output_classes = tuple(ops.Tensor for shape in shapes)
+
+      self._pipeline = pipeline.serialize()
+      self._batch_size = batch_size
+      self._num_threads = num_threads
+      self._device_id = device_id
+      self._exec_separated = exec_separated
+      self._prefetch_queue_depth = prefetch_queue_depth
+      self._cpu_prefetch_queue_depth = cpu_prefetch_queue_depth
+      self._gpu_prefetch_queue_depth = gpu_prefetch_queue_depth
+      self._shapes = tuple(tf.TensorShape(shape) for shape in shapes)
+      self._dtypes = tuple(dtype for dtype in dtypes)
+
+      self._structure = structure.convert_legacy_structure(
+        self._dtypes, self._shapes, output_classes)
+
+      if _get_tf_minor_version() == '14':
+        super(_DALIDatasetV2, self).__init__(self._as_variant_tensor())
+      elif _get_tf_minor_version() == '13':
+        super(_DALIDatasetV2, self).__init__()
+      else:
+        raise RuntimeError('Unsupported TensorFlow version detected at runtime. DALIDataset supports versions: 1.13, 1.14')
```

Comment:
```suggestion
        raise RuntimeError('Unsupported TensorFlow version detected at runtime. DALIDataset supports versions: >= 1.13')
```
?

Reply:
I only tested it for these two. >= would imply 1.15 and 2.0.

Maybe we can change it later, when we will test it properly?

Reply:
OK